Modules/PrinterManager/Windows/WindowsPrinterManager.cs

```diff
@@ -131,8 +132,38 @@ public override void Configure(Printer printer, bool verbose = false)
 
             if (string.IsNullOrEmpty(printer.ConfigFile)) return;
 
-            Log.Entry(LogName, "Configuring " + printer.Name);
-            PrintUI($"/Sr /n \"{printer.Name}\" /a \"{printer.ConfigFile}\" m f g p", verbose);
+            if (File.Exists(printer.ConfigFile))
+            {
+                Log.Entry(LogName, "Configuring " + printer.Name);
+                PrintUI($"/Sr /n \"{printer.Name}\" /a \"{printer.ConfigFile}\" m f g p", verbose);
+                PrintUI($"/Sr /n \"{printer.Name}\" /a \"{printer.ConfigFile}\" m f u p", verbose);
+            }
+            else {
+                // We assume the file may have been passed with parameters, eg 'C:\ConfigFile.dat m f u p', in this case, try to see if a valid file can be found.
+                try
+                {
+                    int endOfPath = printer.ConfigFile.ToLower().LastIndexOf(".dat") + 4;
```

Comment:
If the ConfigFile parameter is set to a file with no `.dat` extension (e.g. `"C:\\Conffile.bin m f g p"`) the result will always be `3`. In such a case the splitting in the following two lines will result in `filePath` being set to `C:\`. While the code still works as intended the error message might confuse people as their file does actually exist. Might be good style to check `endOfPath` before proceeding and print a meaningful error message with a in case ".dat" is not found in the string.

Reply:
Ahh thanks for that, I've updated the latest commit to check for the .dat by checking the index length, if it can't see a valid number an error message is logged stating a .dat file cannot be found. 

I was in two minds about checking if it's a .dat on line 135. Not sure how necessary this is?

Reply:
> I was in two minds about checking if it's a .dat on line 135. Not sure how necessary this is?

The check in line 135 makes a lot of sense to me. While running the PrintUI commands with extra variables in `printer.ConfigFile` would not fail in a terrible way it's not good style and on top would produce miss-leading messages in the logs ([1](https://github.com/FOGProject/fog-client/blob/9755290de68ff6eac3e1904770786be2284845a4/Modules/PrinterManager/Windows/WindowsPrinterManager.cs#L47), [2](https://github.com/FOGProject/fog-client/blob/9755290de68ff6eac3e1904770786be2284845a4/Modules/PrinterManager/Windows/WindowsPrinterManager.cs#L134)).

Reply:
I agree, it has been updated in the latest commit to check first and not do anything if a .dat cannot be found. I think this is the most robust solution.